src/panels/calendar/ha-panel-calendar.ts

```diff
@@ -0,0 +1,304 @@
+                      @change=${this._handleToggle}
+                    >
+                      ${selCal.calendar.name}</paper-checkbox
+                    >
+                  </div>`
+              )}
+            </ha-card>
+          </div>
+          <ha-full-calendar
+            .events=${this._events}
+            @view-changed=${this._handleViewChanged}
+          ></ha-full-calendar>
+        </div>
+      </app-header-layout>
+    `;
+  }
+
+  private _fetchCalendars() {
+    this.hass.callApi<Calendar[]>("GET", "calendars").then((result) => {
+      this._calendars = result.map((cal, idx) => ({
+        selected: true,
+        calendar: cal,
+        backgroundColor: `#${palette[idx]}`,
+      }));
+    });
+  }
+
+  private _fetchData() {
+    if (!this._start || !this._end) {
+      return;
+    }
+
+    if (!this._firstUpdate && this._calendars.length) {
+      this._firstUpdate = true;
+    }
+
+    const start = new Date(this._start);
+    const end = new Date(this._end);
+    const params = encodeURI(
+      `?start=${start.toISOString()}&end=${end.toISOString()}`
+    );
+
+    const calEvents: CalendarEvent[] = [];
+    const calls = new Promise((resolve) => {
+      this._calendars
+        .filter((selCal) => selCal.selected)
+        .forEach(async (selCal, idx) => {
+          const events = await this.hass.callApi<any[]>(
+            "GET",
+            `calendars/${selCal.calendar.entity_id}${params}`
+          );
+          events.forEach((ev) => {
+            const eventStart = this._getDate(ev.start);
+            if (!eventStart) {
+              return;
+            }
+            const eventEnd = this._getDate(ev.end);
+            const event: CalendarEvent = {
+              start: eventStart,
+              end: eventEnd,
+              title: ev.summary,
+              summary: ev.summary,
+              backgroundColor: selCal.backgroundColor,
+              borderColor: selCal.backgroundColor,
+            };
+
+            calEvents.push(event);
+
+            if (idx === this._calendars.filter((c) => c.selected).length - 1) {
+              resolve();
+            }
+          });
+        });
+    });
+
+    calls.then(() => {
+      this._events = calEvents;
+    });
+  }
+
+  private _getDate(dateObj: any): string | undefined {
+    if (typeof dateObj === "string") {
+      return dateObj;
+    }
+
+    if (dateObj.dateTime) {
+      return dateObj.dateTime;
+    }
+
+    if (dateObj.date) {
+      return dateObj.date;
+    }
+
+    return undefined;
+  }
+
+  private _handleToggle(ev) {
+    const cals = this._calendars.map((cal) => {
+      if (ev.target.value === cal.calendar.entity_id) {
+        cal.selected = ev.target.checked;
+      }
+      return cal;
+    });
+    this._calendars = cals;
+    this._fetchData();
+  }
+
+  private _handleViewChanged(ev: HASSDomEvent<CalendarViewChanged>) {
+    this._start = ev.detail.start;
+    this._end = ev.detail.end;
+    this._fetchData();
```

Comment:
Probably need a fetching flag to not fetch data when its fetching already

Reply:
Actually we need to fetch the data because the data it was fetching is now outdated